Lib/pdb.py

```diff
@@ -1532,7 +1532,7 @@ def _runmodule(self, module_name):
         __main__.__dict__.update({
             "__name__": "__main__",
             "__file__": self.mainpyfile,
-            "__package__": module_name,
+            "__package__": mod_spec.parent,
```

Comment:
Huh, I initially thought this was wrong, but I'd forgotten that it's `runpy._get_module_details` that handles the recursive search for `pkg.__main__`, so `mod_spec.parent == module_name` in those cases.